# ADD - Insert node routing table

blackironj opened the pull request:
#### 추가사항
- FindNode Rpc 요청을 받았을 때, 요청자를 자신의 Routing table에
삽입하도록함.

 #### 수정사항
- Netplugin rpc server가 실행 될 때, 확인 하고 있어야 하는 주소는 하나가
아닌 모두 이므로 `0.0.0.0` 으로 수정

 ##### 기타 수정
- certificate handler 에서 certificate를 검증하는 부분을 tethys-utill의
CertVerifier를 사용하여 하도록 수정

morishjs commented on src/plugins/net_plugin/net_plugin.cpp:
```diff
@@ -86,7 +86,10 @@ class NetPluginImpl {
 
-    builder.AddListeningPort(p2p_address, grpc::InsecureServerCredentials());
+    auto [_, port] = getHostAndPort(p2p_address);
+    string listening_addr = "0.0.0.0" + port;
```
port 가 3000 이라면

0.0.0.03000 이 될 것 같은데요.
':'가 빠진 것 같습니다.

blackironj replied:
에구 확인 감사합니다. 수정했어요.

Commit:
# ADD - Insert peer in routing table

 #### 추가사항
- FindNode Rpc 요청을 받았을 때, 요청자를 자신의 Routing table에
삽입하도록함.

 #### 수정사항
- Netplugin rpc server가 실행 될 때, 확인 하고 있어야 하는 주소는 하나가
아닌 모두 이므로 `0.0.0.0` 으로 수정

Commit:
# REFACTOR - Contract / Cert handler

- certificate handler 에서 certificate를 검증하는 부분을 tethys-utill의
CertVerifier를 사용하여 하도록 수정
